catalog/core/catalog-core-attributeregistry/src/main/java/ddf/catalog/data/impl/AttributeRegistryImpl.java

```diff
@@ -43,27 +44,77 @@ public boolean register(final AttributeDescriptor attributeDescriptor) {
 
         final String name = attributeDescriptor.getName();
 
-        if (attributeMap.putIfAbsent(name, attributeDescriptor) == null) {
+        if (attributeMap.containsKey(name)) {
+            AttributeValueType attributeValueType = attributeMap.get(name);
+            AttributeDescriptor comparableAttributeDescriptor =
+                    attributeValueType.getAttributeDescriptor();
+            if (comparableAttributeDescriptor.equals(attributeDescriptor)) {
+                attributeValueType.incrementOccurrences();
```

Comment:
Wouldn't you have to increment it in either case? If an attribute that only matches by name is registered, the counter won't be increased. However, when that attribute is deregistered, the counter will be decreased because `deregister()` only cares that the name matches.